app/models/provider.rb

```diff
@@ -63,4 +63,20 @@ def refresh_ems(opts = {})
     end
     managers.flat_map { |manager| EmsRefresh.queue_refresh(manager, nil, opts) }
   end
+
+  def self.destroy_queue(ids)
+    find(Array.wrap(ids)).each(&:destroy_queue)
```

Comment:
You should not need the Array.wrap...

```ruby
Vm.find(21000000005058, 21000000005063, 21000000005064).size
# => 3
Vm.find([21000000005058, 21000000005063, 21000000005064]).size
# => 3
```

Unless there's some other edge condition I'm not seeing?

Reply:
modeled after [ext_management_system](https://github.com/jameswnl/manageiq/blob/f62dae6ec44b686d1842d117248e28d818c6b69a/app/models/ext_management_system.rb#L443-L443)
I will remove `Array.wrap` then.